### DISCUSSION TITLE
Next big feature poll #3

### DISCUSSION THREAD

**Discussion by seerge** [seerge/g-helper#583] [Polls]
Hello, just wanted to run a poll to gather opinions of what next big feature would be most helpful.

**TOP 2 features will have best chances to get implemented**

**Comment by Moorole** [comment]
undervolting amd cpu also?

**Reply by seerge** [reply]
Only AMD

**Reply by RithikGobinath** [reply]
not for intel?

**Reply by seerge** [reply]
Not for intel :)

**Reply by dailac-boiii** [reply]
> not for intel?

you can use XTU/ThrottleStop for that

**Reply by Metalogic999** [reply]
I didn't think you could undervolt AMD "HS" CPUs like the R9 7940HS? Only the GPU?

**Reply by seerge** [reply]
@Metalogic999 i also wonder, try build from next comment https://github.com/seerge/g-helper/discussions/583#discussioncomment-6200024
(and check logs, it puts undervolting results there if they were accepted or not)

**Reply by Metalogic999** [reply]
@seerge I'll give it a go, I've already successfully undervolted the GPU to 850mV on my G14 using MSI Afterburner, will see if undervolting the 7940HS CPU works with G-Helper

**Reply by Mikealll** [reply]
![image](https://github.com/seerge/g-helper/assets/136940829/7d680b30-f45c-409d-8de1-81c52c3772d4)
@Metalogic999 I believe you can only undervolt Ryzen 9 HS H and HX my 6800HS fails both in my flow x13.

**Reply by seerge** [reply]
@Mikealll that's a pity, but i understand undervolting doesn't work for you using other apps too like uxtu ?

**Comment by ke1ne** [comment]
I'm voting for CPU Undervolting because of even better temperatures possibility. But implemented in simple enough way, tweaking only basic values.

For the always-in-beta-geeky-control-freaky way, some tools are already exist. Say No to another Armory Crate! =))
![image](https://github.com/seerge/g-helper/assets/8934421/0dc93fe4-32c4-435f-9fdd-452475353319)

**Reply by seerge** [reply]
Thats a wrong tab on your screenshot, undervolting is in different one :))

**Reply by ke1ne** [reply]
> Thats a wrong tab on your screenshot, undervolting is in different one :))

Just an example of a million controls nightmare! :>

**Reply by seerge** [reply]
For undervolting you need only one - it's called "all core optimization"

**Reply by ke1ne** [reply]
> For undervolting you need only one - it's called "all core optimization"

Sounds like magic. If you decide to implement, ready to test.

**Reply by NJinsei** [reply]
> For undervolting you need only one - it's called "all core optimization"

Isn't there also one for the IGPU as well ?

**Reply by seerge** [reply]
@ke1ne @NJinsei 
You can try a build from next comment https://github.com/seerge/g-helper/discussions/583#discussioncomment-6200024

**Comment by GeometricParrot** [comment]
Is there a way to map a macro to one of the 'M' keys? Something like Control+C or Win+A. I could get a single key working in G-helper, but not multiple.

**Reply by seerge** [reply]
Currently no, but you can map M to F13 - F24 keys -> and then reuse them in Autohotkey for macro of any complexity

**Reply by GeometricParrot** [reply]
Thank you, that works well.

**Comment by R4333** [comment]
I have a request. If you end up implementing saving custom profiles, could you make it so that people can share the config files which would work on the same model of the laptop. It would make it quite easy for the folks on the /r/ZephyrusG14 to share custom configs with others. I am sure everyone would like that.

**Reply by bel57** [reply]
+1 would it be possible to create a builtin community hub with a list of profiles available according to the detected model?

**Reply by TomBonnot** [reply]
+1 but @bel57 I think that reddit or maybe a discussion onto github would be enough, I mean, it's a lightweight app.

**Reply by bel57** [reply]
By community hub what I meant is just space containing profiles and maybe the ability to vote. Or have the hub here on Github and add the best profiles to G-Helper so user can select custom profiles without further steps.

**Reply by seerge** [reply]
Well, you can share your config already by just taking it from ``%appdata%/ghelper``, aren't you ?

**Comment by seerge** [comment]
First beta with Custom Profiles : 

### [GHelper.zip](https://github.com/seerge/g-helper/files/11714720/GHelper.zip)

Feedback is welcome
![Screenshot 2023-06-11 152206](https://github.com/seerge/g-helper/assets/5920850/ddb60507-017d-48ba-bd5f-0c6fb57ceb06)

**Reply by gettingerr** [reply]
Working great! Would be cool to be able to cycle to these custom profiles using the Macro-Keys.

Also renaming would be helpful so it's easy to know what you're picking. Thanks so much for all your hard work it's awesome.

**Reply by ke1ne** [reply]
> First beta of Custom Profiles : [GHelper.zip](https://github.com/seerge/g-helper/files/11713428/GHelper.zip)
> 
> Feedback is welcome

Seems to work well. Thank you!

**Reply by seerge** [reply]
@gettingerr i have intentionally excluded custom profiles from hotkey cycling (as you can theoretically have like 20 profiles) :)

**Reply by NJinsei** [reply]
Does this work with custom power plans as well ?

**Reply by seerge** [reply]
@NJinsei if you mean "power modes" (i.e. windows battery saver / max performance) it will just copy what was used in Silent/Balanced/Turbo .

For custom "power plans" (that you set via config) - you can use same approach, and add any new power plan GUIDs for new modes as well.

**Reply by gettingerr** [reply]
> @gettingerr i have intentionally excluded custom profiles from hotkey cycling (as you can theoretically have like 20 profiles) :)

Would it be possible (without weighing the ui down too much) to users to select certain profiles to be included?

**Reply by seerge** [reply]
Added renaming and FN+F5 should include all custom modes in a loop (and updated build in head-comment too)
[GHelper.zip](https://github.com/seerge/g-helper/files/11714646/GHelper.zip)

**Reply by justing6** [reply]
I also would vote to let the end user customize this behavior, personally I would only want FN-F5 to toggle between the original 3 modes and just select the custom ones from dropdowns when I need them

**Comment by bowes332frank** [comment]
Good Game.

**Comment by Jokerr2** [comment]
Can you change it to where if someone switches to eco mode and does restart the GPU, it stays disabled in the device manager and doesn't get enabled again? And when switching back to the standard mode it enables it again maybe from device manager? or add a button to shutdown discrete GPU or smth

**Reply by seerge** [reply]
It shouldn't switch back to standard by itself, but I heard that some models (like G15) have bugged bios that can misbehave. You can try to experiment with turning fast boot ON or OFF (in bios - F2 on boot)

**Comment by G-rox-y** [comment]
Would be cool if when prompted with "download update" it would be possible to see what's new in that update

**Reply by seerge** [reply]
You can always read what's new in releases by clicking underlined  app version

**Reply by G-rox-y** [reply]
Oh, i didn't notice that, cool!

**Comment by seerge** [comment]
First beta with **Undervolting** 

### [GHelper.zip](https://github.com/seerge/g-helper/files/11776144/GHelper.zip)

To apply Undervolting you need to manually click Apply first, and if it's stable - you can check "Auto Apply" next to it

![Screenshot 2023-06-16 234502](https://github.com/seerge/g-helper/assets/5920850/365a0530-53da-4559-b9cc-6e27347e1b78)

**Reply by Jokerr2** [reply]
just tested it with different values, if you have amd ryzen 9 6900hs i would not recommend going below the values shown in the screenshot above otherwise the igpu basically just obliterates out of existence and your display drivers get fucked and you gotta shut down and press power button for 15 sec before the display drivers reset or smth, even if you do use the values shown above in the screenshot, its too unstable. i didn't notice any significant improvement in battery life, it was basically the same (I mainly did battery test by surfing on my browser, might improve battery life when it comes to more higher work load). if you get different results let me know, maybe i am just dumb and dont know how to properly undervolt a CPU. maybe dont undervolt igpu and just undervolt CPU, because the main thing that's unstable is the display so maybe that might work

**Reply by seerge** [reply]
@Jokerr2 that's how undervolting works, if it's too low - your APU will get unstable. It really depends from unit to unit. But I have limited sliders a bit (updated build in head comment)

CPU : -30 ... 0
iGPU : -20 ... 0

**Reply by AreyUA** [reply]
Well, my situation is the following. with my tasks and the heaviest loads, the processor is loaded on average 6-15%, and the video card up to 35-55%. in terms of temperatures, the processor heats up to 80+, and the video card up to 65. And this is all in the Balance mode, with 15W tgp. I don't know why it gets so hot and what to do

**Reply by ke1ne** [reply]
@seerge Just tried the Turbo and -10 for CPU (Ryzen 9 5900HS). From the "consumer" side, it seems to behave OK. How to test it properly? Like checking some values in HWINFO, etc.

**Reply by Moorole** [reply]
just tested ca. 10 min with prime 95 on -30 with my 4800hs. no issues and according hwinfo i get a nice undervolt applied. just want to run a benchmark as undervolting actually should lead to more cpu strenght as thermal trottle and power throttle will happen later. anyone a suggestion for a good tool to benchmark cpu?
Seerge, thank you so much for all that programming... ::))

**Reply by seerge** [reply]
@ke1ne you should see a bit higher score in cinebench or so, but don’t expect miracles. -10 is a very conservative value:)

**Reply by Jokerr2** [reply]
yeah so after i undervolted only the CPU and left the igpu alone, it became much more stable, i could go to -20 for CPU before it started giving me some problems

**Reply by AreyUA** [reply]
Noticed an interesting thing. In Turbo mode, at -20, the temperature drops and the FPS grows a little, at 30 watts. And if you turn on the Balance mode at 15W from -20, then the temperatures, on the contrary, rise and exceed 75, and sometimes jump up to 80 (((

**Reply by ke1ne** [reply]
> @ke1ne you should see a bit higher score in cinebench or so, but don’t expect miracles. -10 is a very conservative value:)

@seerge Tested with the latest Cinebench with only CPU undervolted:

![image](https://github.com/seerge/g-helper/assets/8934421/40832987-7c8c-4bec-9e7e-c8d46ed1a1a6)

**Reply by seerge** [reply]
@ke1ne not much of a gain :) is it even working for you? (check log in %appdata%/ghelper it reports status there)

**Reply by ke1ne** [reply]
> @ke1ne not much of a gain :) is it even working for you? (check log in %appdata%/ghelper it reports status there)

Got this when undervolted CPU and set back to zero:
```
17.06.2023 11:21:06: UV: -30 OK
17.06.2023 11:21:06: iGPU UV: 0 CMD_REJECTED_PREREQ
17.06.2023 11:21:38: UV: 0 OK
17.06.2023 11:21:39: iGPU UV: 0 CMD_REJECTED_PREREQ
```

P.S. Also, it spams the log A LOT with this:
```
17.06.2023 11:06:16: Key: 236
17.06.2023 11:06:24: Key: 236
17.06.2023 11:06:35: Key: 236
17.06.2023 11:06:48: Key: 236
17.06.2023 11:07:05: Key: 236
17.06.2023 11:07:18: Key: 236
17.06.2023 11:08:02: Key: 236
17.06.2023 11:08:53: Key: 236
17.06.2023 11:09:01: Key: 236
17.06.2023 11:09:31: Key: 236
17.06.2023 11:10:47: Key: 236
17.06.2023 11:11:19: Key: 236
```

**Reply by seerge** [reply]
@ke1ne 236 is a "fake" key keyboard spams sometimes, i have suppressed that log message in 0.87. As for undervolting - it seem to get accepted, so it's good

**Reply by Metalogic999** [reply]
@seerge where do I need to copy the DLL files to that come with G-Helper 0.87?

**Reply by seerge** [reply]
@Metalogic999 same folder as exe

**Reply by Moorole** [reply]
Got an improvement from 10053 to 10150. but changing from the beta to 0.87 I get following mistake everytime I apply the undervolt. I put ghelper on desktop with the other two files![image](https://github.com/seerge/g-helper/assets/133746822/ec339f5b-e1cc-4fd6-8665-e8477000ce3e)

Edit:Disappears with the DLL in the folder

Would it be possible to open to -50 (is it mv), I feel like I didn't reach the limit of my silicone.

**Reply by NJinsei** [reply]
Pretty sure for these laptop APU's -30 CPU is the limit

**Reply by seerge** [reply]
@Moorole looks like you don't have that winring dll in a same folder as exe (or post a full error text, as it's impossible to read)

**Reply by Moorole** [reply]
> Pretty sure for these laptop APU's -30 CPU is the limit

Well I see no problems with -30 and with undervolting I don't think there is a danger anyway. So I'd like to go as low as possible. No reason to stop half way😉

**Reply by schm1dtmac** [reply]
Will installing the WinRing0 driver be mandatory from now on for using G-Helper btw, or will it be optional and only needed to access the undervolting feature?

**Reply by seerge** [reply]
@schm1dtxbox atm it's needed only for undervolting, as there is no way to do it w/o that. And you don't need to "install" it anywhere, just unzip into same folder.

**Reply by seerge** [reply]
@Moorole i will keep it at -30 for now, as my silicone crashes even on -16. I just predict a wave of strange topics on reddit from people setting something inadequate and then immediately  blaming g-helper for "crashing their laptop" or so...

**Reply by ke1ne** [reply]
> @Moorole i will keep it at -30 for now, as my silicone crashes even on -16. I just predict a wave of strange topics on reddit from people setting something inadequate and then immediately blaming g-helper for "crashing their laptop" or so...

Nobody reads the readme.md, that's true. =)

**Reply by seerge** [reply]
Just in case, Rog ALLY refuses to undervolt , at least using current approach :) (if someone was wondering)

**Reply by ke1ne** [reply]
> @ke1ne 236 is a "fake" key keyboard spams sometimes, i have suppressed that log message in 0.87. As for undervolting - it seem to get accepted, so it's good

Sorry, still spams with the "Key: 236"

**Reply by seerge** [reply]
You need to download 0.87 from releases, beta from this thread is not actual anymore

**Reply by ke1ne** [reply]
> You need to download 0.87 from releases, beta from this thread is not actual anymore

Sorry, I'm talking about 0.87

**Comment by Metalogic999** [comment]
> @Metalogic999 same folder as exe

Thanks

**Comment by Kickskii** [comment]
if youre adding undervolting i request that you add it in a toggle which allows for per core vs all core CO
i have one core that is holding every other core back in terms of doing curve optimizer 
just a suggestion

**Reply by TexYukino** [reply]
Nice idea. And I want to ask that if i undervolt and my pc crashes, is it just simply reboot my pc and do no harm(except software damage)?

**Reply by seerge** [reply]
Yes, after rebooting all your settings are reset anyway. That's why undervolting has separate Apply and Auto Apply boxes. You need to test if it's stable with Apply first.

**Reply by Kickskii** [reply]
if you're using ryzenadj for perco you can check out excel sheets i wrote down with values that reflect the curve 
maybe this is helpful
the command for ryzenadj for running a per core is ryzenadj.exe --set-coper=0x0FFFEC 
[Curve Optimizer Look up table.xlsx](https://github.com/seerge/g-helper/files/11804850/Curve.Optimizer.Look.up.table.xlsx)
in case you want to check values heres a calculation excel i made that updates values in real time and has explanations on the formula thats used 
[the calculation chart for nerds.xlsx](https://github.com/seerge/g-helper/files/11804873/the.calculation.chart.for.nerds.xlsx)
overall 
basically this in ryzenadj
![image](https://github.com/seerge/g-helper/assets/56576955/078641f2-d2ea-4adc-a9ee-693025532ee4)
ive been using a batch file to run percore since november and if its added to ghelper itll just be a more cohesive way of doing it since i wouldnt need the task scheduler thing anymore

edit: heres an example of how the batch file i use looks like in case you're interested 
![image](https://github.com/seerge/g-helper/assets/56576955/21d56bd7-3c3d-4568-92f1-1a1f5f29650a)
